File: sp/server/game/tile.go
package game

type Tile struct {
	Row         int    `json:"row"`
	Column      int    `json:"column"`
	Set         bool   `json:"set"`
	Highlighted bool   `json:"highlighted"`
	Type        int    `json:"type"`
	Letter      Letter `json:"letter"`
}

func (tile Tile) getWordMultiplicand() int {
	wordMultiplicand := 1
	if tile.Type == MULTIPLY_WORD_2 {
		wordMultiplicand = 2
	} else if tile.Type == MULTIPLY_WORD_3 {
		wordMultiplicand = 3
	}
	return wordMultiplicand
}

func (tile Tile) getTilePoints() int {
	tilePoints := tile.Letter.Points
	tileMultiplicand := 1
	if tile.Type == MULTIPLY_LETTER_2 {
		tileMultiplicand = 2
	} else if tile.Type == MULTIPLY_LETTER_3 {
		tileMultiplicand = 3
	}
	tilePoints *= tileMultiplicand
	return tilePoints
}

File: sp/client/src/main/kotlin/model/game/Tile.kt
package model.game

import com.beust.klaxon.Json

// a class representing a KrisKros tile
class Tile(val row: Int, val column: Int, val type: Int, var selected: Boolean = false, var highlighted: Boolean = false, var letter: Letter? = null, var set: Boolean = false) {
    @Json(ignored = true)
    val typeEnum = if (type < TileType.values().size) TileType.values()[type] else TileType.BASIC
}

enum class TileType {
    BASIC,
    MULTIPLY_WORD_2,
    MULTIPLY_WORD_3,
    MULTIPLY_LETTER_2,
    MULTIPLY_LETTER_3
}

fun Tile.getPoints(letter: Letter): Int =
        when (this.typeEnum) {
            TileType.MULTIPLY_LETTER_2 -> letter.points * 2
            TileType.MULTIPLY_LETTER_3 -> letter.points * 3
            else -> letter.points
        }